### DISCUSSION TITLE
Doubts on 'unstable_getServerSession' and 'useSession'

### DISCUSSION THREAD

**Discussion by FabioDainese** [nextauthjs/next-auth#5655] [Help]
Hi guys, I'm kinda new to *next-auth* and I've some doubts on when to use `useSession(...)` and `unstable_getServerSession(...)`. Suppose you have to display different content of a page based on the fact that a user has logged in or not, I would do something like this:

```javascript
export async function getServerSideProps(context) {
    return {
        props: {
            auth_session: await unstable_getServerSession(context.req, context.res, authOptions)
        }
    }
}

export default function Page({ auth_session }) {
    return (
        <>
            {auth_session ? (
                <p>Hi {auth_session.user.name}</p>
            ) : (
                <button onClick={() => signIn('google', { callbackUrl: '/' })}>Log in with Google</button>
            )}
        </>
    )
}
```

But looking at the [documentation](https://next-auth.js.org/tutorials/securing-pages-and-api-routes#server-side) I don't get it why I also need to use the `useSession(...)` hook if I'm already passing the data with the `getServerSideProps(...)`. Is there a particular reason for that? Plus, doesn't that also means adding an additional fetch (for getting the same information)?

**Comment by ThangHuuVu** [comment]
`unstable_getServerSession` is specific to the server, while `useSession` is a React Context hook. Normally, you want this hook if you are for example accessing the session object in a nested component layout.
